src/main/scala/vsys/blockchain/state/opcdiffs/AssertOpcDiff.scala

```diff
@@ -70,34 +72,47 @@ object AssertOpcDiff {
       Right(OpcDiff.empty)
   }
 
-  object AssertType extends Enumeration {
-    val GteqZeroAssert = Value(1)
-    val LteqAssert = Value(2)
-    val LtInt64Assert = Value(3)
-    val GtZeroAssert = Value(4)
-    val EqAssert = Value(5)
-    val IsCallerOriginAssert = Value(6)
-    val IsSignerOriginAssert = Value(7)
+  def checkHash(hashValue: DataEntry, hashKey: DataEntry): Either[ValidationError, OpcDiff] = {
+    if (hashValue.dataType != DataType.ShortText || hashKey.dataType != DataType.ShortText)
+      Left(ContractDataTypeMismatch)
```

Comment:
need else after this line

Reply:
fixed in last commit, I think I should use return